Includes/networkManager.h

```diff
@@ -0,0 +1,47 @@
+#ifndef NEVOLUTIONX_NETWORKMANAGER_H
+#define NEVOLUTIONX_NETWORKMANAGER_H
+
+#include "config.hpp"
+
+#include <thread>
+
+class NetworkManager {
+public:
+  typedef enum {
+    INIT_NONE,
+    INIT_SUCCEEDED,
+    INIT_FAILED,
+    RUNNING
+  } NetworkStatus;
+
+public:
+  explicit NetworkManager(Config config);
+
+  // Initializes the network asynchronously.
+  void asyncInit();
+
+  inline NetworkStatus status() const { return status_; }
+  std::string IPAddressString() const;
+
+  inline bool isNewlyInitialized() {
+    if (status_ == INIT_SUCCEEDED) {
+      status_ = RUNNING;
+      init_thread_.join();
+      return true;
+    }
+
+    return false;
+  }
+
+private:
+  static void thread_main(NetworkManager *manager);
+
+private:
+  Config config_;
```

Comment:
Repository doesn't use `_` suffix anywhere outside this file.

Reply:
Dropped